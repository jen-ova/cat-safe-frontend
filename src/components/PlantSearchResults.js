import React from "react";
import Plant from "./Plant";
import "../styles/plantSearchResults.css";

const PlantSearchResults = ({ results, hasSearched, filtered }) => {
  console.log(filtered);

  if (results.length === 0 && hasSearched === false) {
    return null;
  }
  if (results.length === 0 && hasSearched === true) {
    return (
      <p className="search-results__no-results">
        Sorry, no results exist for this search! Please try again.
      </p>
    );
  } else {
    return (
      <div>
        {filtered.map((plant) => (
          <Plant
            key={plant.plantName}
            name={plant.plantName}
            scientificName={plant.scientificName}
            image={plant.link}
          />
        ))}
      </div>
    );
  }
};

export default PlantSearchResults;
