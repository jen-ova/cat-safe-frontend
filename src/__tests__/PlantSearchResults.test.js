import React from "react";
import { render } from "@testing-library/react";
import PlantSearchResults from "../components/PlantSearchResults";

describe("Search", () => {
  const { asFragment } = render(<PlantSearchResults />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
