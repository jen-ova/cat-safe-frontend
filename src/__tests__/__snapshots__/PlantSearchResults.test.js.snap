// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div>
    <p>
      Search results go here
    </p>
  </div>
</DocumentFragment>
`;
